components/mgmtworker/scripts/start.py

```diff
@@ -10,7 +11,44 @@
 import utils  # NOQA
 
 MGMT_WORKER_SERVICE_NAME = 'mgmtworker'
+CELERY_PATH = '/opt/mgmtworker/env/bin/celery'  # also hardcoded in create
+RETRIES = 10
+
+ctx_properties = utils.ctx_factory.get_install_properties(
+    MGMT_WORKER_SERVICE_NAME)
+
+
+def is_worker_running(amqp_url):
+    """Use `celery status` to check if the worker is running."""
+    resp = utils.sudo([
+        CELERY_PATH,
+        '-b', celery_amqp_url,
+        '--app=cloudify_agent.app.app',
+        'status'
+    ], ignore_failures=True)
+    return resp.returncode == 0
 
 
 ctx.logger.info('Starting Management Worker Service...')
 utils.start_service_and_archive_properties(MGMT_WORKER_SERVICE_NAME)
+
+if utils.systemd.is_alive(MGMT_WORKER_SERVICE_NAME):
+    ctx.logger.info('Management Worker Service is running')
+else:
+    ctx.logger.error('Management Worker Service is not running')
```

Comment:
again, if a service is not running, the operation should be aborted. There's no situation in which the service is not running (after retries maybe) and it's ok.